index.js

```diff
@@ -112,6 +122,52 @@ class Conf {
 		}
 	}
 
+	_migrate(migrations, versionToMigrate = '0.0.0') {
+		const MIGRATION_KEY = '__conf-migrated-version__';
+
+		const previousMigratedVersion = this.get(MIGRATION_KEY, '0.0.0');
+
+		const newerVersions = Object.keys(migrations)
+			.filter(candidateVersion => this._shouldPerformMigration(candidateVersion, previousMigratedVersion, versionToMigrate))
+			.sort(semver.compare);
+
+		newerVersions.forEach(version => {
+			try {
+				const migration = migrations[version];
+				migration(this);
+
+				this.set(MIGRATION_KEY, version);
```

Comment:
Small remark: because we are setting the version here, we would have a duplicate set operation on line 147 for the last entry of the `newerVersions` array. Moreover, if there `newerVersions` array is empty, the set operation on line 147 will always run.

To avoid both issues we can:

1. Set `previousMigratedVersion = version`  here and add a condition on line 147 that only performs the set if `previousMigratedVersion !== versionToMigrate`
2. Only perform this set inside the catch and add a condition on line 147 that only performs the set if `previousMigratedVersion !== versionToMigrate`

I prefer `1` because it's a little bit safer than `2`, specially on unhandled crashes (e.g.: OOM), but has a bit more I/O.

Reply:
I've left the `set` in the last line for the following case:

(Assume the last time we ran the migrations it was on version 1.0.0)
```js
const migrations = {
    '1.0.0': () => initialMigration,
    '1.0.1' : () => intermediateMigration,
    '1.0.2': () =>  lastMigration
}
```

and let's say we are on version `1.0.3`

If the next time we add a migration, e.g.: 
```js
const newMigrations = {
    '1.0.0': () => initialMigration,
    '1.0.1' : () => intermediateMigration,
    '1.0.2': () =>  lastMigration,
    '1.0.3': () =>  actuallyLastMigration,
}
```
(and the version is still 1.0.3), do we want the `actuallyLastMigration` to run? 

I assumed that we want the same as we discussed previously.

HOWEVER, one could argue that in this case, it makes sense. Because although we didn't change versions (1.0.3 -> 1.0.3), we still would want the migration code from a set of versions that weren't matched before (1.0.2 -> 1.0.3) 🤔

Reply:
I think that my suggestion is unrelated to what you described. The change I suggested is to avoid having to do `this.set(MIGRATION_KEY, versionToMigrate);` if the value is already that: `versionToMigrate === previousMigratedVersion`.

Reply:
You're right, sorry about that.

Added the semver's comparision function, just in case,

`semver.eq(previousMigratedVersion, versionToMigrate)`.

Reply:
You're right, sorry about that.

Added the semver's comparision function, just in case,

`semver.eq(previousMigratedVersion, versionToMigrate)`.